### ISSUE TITLE
[BUG] Error on running  Connect-PnPOnline using credentials

### ISSUE THREAD

**Issue by lipalath-ms** [pnp/powershell#1625]
I have the following script to connect to PnP:
```  
    $sharePointTenantName = $secureSharePointTenantName.SecretValueText
    $sharePointUSSiteUrl = "https://$sharePointTenantName.sharepoint.com"
    $sharePointUSTenantUrl = "https://$sharePointTenantName-admin.sharepoint.com"

    $secureUsername = Get-AzKeyVaultSecret -VaultName $keyVaultName -Name "sharePointTenantAdminUsername"
    $securePassword = Get-AzKeyVaultSecret -VaultName $keyVaultName -Name "sharePointTenantAdminPassword"
    $credential = New-Object System.Management.Automation.PSCredential($secureUsername.SecretValueText, $securePassword.SecretValue)   

   Install-ModuleIfNeeded -Name SharePointPnPPowerShellOnline -Version "3.22.2006.2" -Verbose

    $url = $sharePointUSTenantUrl
    if($ConnectionType -ne "tenant")
    {
        $url = $sharePointUSSiteUrl
        $url = "$url/$ConnectionType/$SiteName"
        Connect-PnPOnline -Url $url -Credentials $credential
    }
    else
   {
        Connect-PnPOnline -Url $sharePointUSSiteUrl -Credentials $credential -TenantAdminUrl $sharePointUSTenantUrl
    }
```
I see this error: "The sign-in name or password does not match one in the Microsoft account system". I verified the credentials and those are correct. What am I missing?

### Expected behavior 
The package should be uploaded/published to app catalog

### Actual behavior 
![image](https://user-images.githubusercontent.com/6519992/153641375-f4a6e4dd-c45a-488b-8f89-68fa87de8467.png)

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)

### What is the version of the Cmdlet module you are running?
No values ![image](https://user-images.githubusercontent.com/6519992/153641612-9dfff520-f046-4b8b-988a-f94cc51131da.png)

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by veronicageek** [comment]
@lipalath-ms 

> Install-ModuleIfNeeded -Name SharePointPnPPowerShellOnline -Version "3.22.2006.2" -Verbose

You are installing the old module which is deprecated and not maintained anymore.

This repo is for the new PnP.PowerShell module. Please read the documentation to get you started:
https://pnp.github.io/powershell/articles/installation.html

**Comment by lipalath-ms** [comment]
@veronicageek  I used PnP.PowerShell module and I still see the same error. I followed this article: https://www.sharepointdiary.com/2019/03/connect-pnponline-with-appid-and-appsecret.html
and with that I see the error: The remote server returned an error: (401) Unauthorized. On following this article https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azuread, I see this error:
Connect-PnPOnline : Exception has been thrown by the target of an invocation.

What am I missing?
